pkg/storage/store.go

```diff
@@ -318,6 +317,10 @@ func (s *store) SelectSamples(ctx context.Context, req logql.SelectSampleParams)
 	return newSampleBatchIterator(ctx, s.chunkMetrics, lazyChunks, s.cfg.MaxChunkBatchSize, matchers, filter, extractor, req.Start, req.End)
 }
 
+func (s *store) ActivePeriodConfig() chunk.PeriodConfig {
+	return s.activeIndexType
```

Comment:
I think this will be a problem because the active index can change while Loki is running. I think this needs to be made smarter such that every time this method is called the active index is calculated based on the current time, WDYT?

Reply:
There are other configs that we tune based on whether the active index is boltdb shipper or not. We will have to take care of all of them. I will see if it is doable without adding a lot of overhead otherwise I think it would be good to ask users to restart ingesters and queriers once which would be a one time thing.